src/core/dataProviders/dataProvider.js

```diff
@@ -266,8 +268,59 @@ export default () => {
      */
     return (type, resource, params) => {
         let { url, options } = convertDataRequestToHTTP(type, resource, params);
-        return fetch(url, options).then(response => response.json())
-            .then(res => convertHTTPResponse(res, type, resource, params))
-            .catch(err => console.log('Error: ', err));
+        if (resource === `patients`) {
```

Comment:
Instead of adding that to the main dataProvider, I advise you to *decorate* the main data provider, as follows:

```js
const dataProvider = (type, resource, params) => { /* the existing code */ };
const fakePatientsProvider = (type, resource, params) => { /* the fake patients code */ };
export default (type, resource, params) => {
   if (resource == 'patients') {
      return fakePatientsProvider(type, resource, params);
  }
  return dataProvider(type, resource, params);
}
```

That way the main data provider remains agnostic of the resource, and doesn't grow too much.

Reply:
Done